src/app/clients/ActionBar/ActionsModal.tsx

```diff
@@ -0,0 +1,171 @@
+import FreeFormTextInput from "@/components/DataInput/FreeFormTextInput";
+import { DatePicker, DesktopDatePicker } from "@mui/x-date-pickers";
+import DropdownListInput from "@/components/DataInput/DropdownListInput";
+import supabase from "@/supabaseClient";
+
+interface ActionsModalProps extends React.ComponentProps<typeof Modal> {
+    data: ClientsTableRow[];
+    errorText: string | null;
+    inputComponent?: React.ReactElement;
+    showSelectedParcels: boolean;
+}
+
+const Centerer = styled.div`
+    display: flex;
+    justify-content: center;
+`;
+
+const Heading = styled.div`
+    font-size: 1.2rem;
+    margin: 0.3rem;
+`;
+
+const ModalInner = styled.div`
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+    align-items: stretch;
+`;
+
+const StatusText = styled.p`
+    margin-left: 1rem;
+    border-top: 1px solid ${(props) => props.theme.main.lighterForeground[2]};
+    padding: 1rem 0;
+    &:last-child {
+        border-bottom: 1px solid ${(props) => props.theme.main.lighterForeground[2]};
+    }
+`;
+
+interface DriverOverviewInputProps {
+    onDateChange: (newDate: Dayjs | null) => void;
+    onDriverNameChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export const DriverOverviewInput: React.FC<DriverOverviewInputProps> = ({
+    onDateChange,
+    onDriverNameChange,
+}) => {
+    return (
+        <>
+            <Heading>Delivery Information</Heading>
+            <FreeFormTextInput onChange={onDriverNameChange} label="Driver's Name" />
+            <DatePicker onChange={onDateChange} disablePast />
+        </>
+    );
+};
+
+interface DayOverviewInputProps {
+    onDateChange: (newDate: Dayjs | null) => void;
+    onCollectionCentreChange: (event: SelectChangeEvent) => void;
+    setCollectionCentre: React.Dispatch<React.SetStateAction<string>>;
+}
+
+export const DayOverviewInput: React.FC<DayOverviewInputProps> = ({
+    onDateChange,
+    onCollectionCentreChange,
+    setCollectionCentre,
+}) => {
+    const [collectionCentres, setCollectionCentres] = useState<[string, string][] | null>(null);
+
+    useEffect(() => {
+        (async () => {
+            const { data, error } = await supabase
+                .from("collection_centres")
+                .select("primary_key, name");
+            if (error) {
+                throw Error(
+                    "We were unable to fetch the collection centre data. Please try again later"
+                );
+            }
+
+            const transformedData: [string, string][] = data.map((item) => [
+                item.name,
+                item.primary_key,
+            ]);
+            setCollectionCentres(transformedData);
+            setCollectionCentre(transformedData[0][1]);
+        })();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    return (
+        <>
+            <Heading>Location</Heading>
+            {collectionCentres !== null && (
```

Comment:
can just change this to !!collectionCentres

Reply:
changed to just `collectionCentres` <3